src/Hosting/Hosting/src/WebHostBuilderExtensions.cs

```diff
@@ -113,20 +113,23 @@ public static IWebHostBuilder Configure(this IWebHostBuilder hostBuilder, Action
         return hostBuilder
             .ConfigureServices((context, services) =>
             {
-                services.AddSingleton(typeof(IStartup), sp =>
+                services.AddSingleton(typeof(IStartup), GetStartupInstance);
+
+                [UnconditionalSuppressMessage("Trimmer", "IL2072", Justification = "Startup type created by factory can't be determined statically.")]
```

Comment:
Why not?

Reply:
Line 131:

```
StartupLoader.LoadMethods(serviceProvider, instance.GetType(), ...);
```

The type parameter on `LoadMethods` is annotated. `instance.GetType()` doesn't have any metadata so it needs to be suppressed.

Reply:
I wonder if we should `TStartup` instead.

Reply:
Discussed more offline and will leave as is